src/util.rs

```diff
@@ -103,6 +103,29 @@ pub fn sha256_digest<R: Read>(input: R) -> Result<String, ActixError> {
         .collect::<String>())
 }
 
+/// Enum representing different strategies for handling spaces in filenames.
+#[derive(Debug, Clone, Copy, serde::Serialize, serde::Deserialize)]
+#[serde(rename_all = "lowercase")]
+pub enum SpaceHandling {
+    /// Represents encoding spaces (e.g., using "%20").
+    Encode,
+    /// Represents replacing spaces with underscores.
+    Replace,
+}
+
+/// Processes the given filename based on the provided space handling strategy.
+///
+/// If `handling` is `Some(SpaceHandling::Encode)`, spaces in the filename are replaced with "%20".
+/// If `handling` is `Some(SpaceHandling::Replace)`, spaces in the filename are replaced with underscores.
+/// If `handling` is `None`, the filename is returned unchanged.
+pub fn process_filename(file_name: &str, handling: Option<SpaceHandling>) -> String {
+    match handling {
+        Some(SpaceHandling::Encode) => file_name.replace(' ', "%20"),
+        Some(SpaceHandling::Replace) => file_name.replace(' ', "_"),
+        None => file_name.to_string(), // keep the filename unchanged
+    }
+}
```

Comment:
You can possibly make this function a part of of the `SpaceHandling` enum.

Reply:
Addressed in [94ae863](https://github.com/orhun/rustypaste/commit/94ae863f61b0b59f4ab9959abc828a39960bb0e4).